src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java

```diff
@@ -0,0 +1,292 @@
+  }
+
+  /**
+   * Check if groupId and artifactId matches with OWASP Dependency GA.
+   *
+   * @param groupId to be checked.
+   * @param artifactId to be checked.
+   * @return True if the OWASP Dependency GA matches, false otherwise.
+   */
+  private static boolean isDependencyCheck(String groupId, String artifactId) {
+    return "org.owasp".equals(groupId) && "dependency-check-maven".equals(artifactId);
+  }
+
+  /**
+   * Check the usage of OWASP Dependency Check plugin in the project.
+   * 
+   * @param locations Set of location hits which indicates where exactly the plugin was found.
+   * @return The usage status of type {@link OwaspDependencyCheckUsage}.
+   */
+  private static OwaspDependencyCheckUsage usageFrom(Set<Location> locations) {
+    if (locations == null || locations.isEmpty()) {
+      return NOT_USED;
+    }
+
+    if (mandatoryCheckFor(locations)) {
+      return MANDATORY;
+    }
+
+    return OPTIONAL;
+  }
+
+  /**
+   * Check if plugin configurations have an attribute which can fail the build on finding
+   * vulnerabilities above a given CVSS score. Hence, look for possible CVSS score value which
+   * indicates, when to fail the build.
+   *
+   * @param configuration The plugin configuration to be checked.
+   * @return The cvss score which indicates how the build should fail. Otherwise null.
+   */
+  private static Double cvssScoreFrom(Object configuration) {
+    // The OWASP Dependency Check plugin is found and checking configurations.
+    for (String name : FAIL_BUILD_CONFIGURATIONS.keySet()) {
+      Optional<String> content = parameter(name, configuration);
+
+      if (content.isPresent()) {
+        String type = FAIL_BUILD_CONFIGURATIONS.get(name);
+        return parseScore(content.get(), type);
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Parse the value depending on the given type. The type can be 'boolean' or 'number'. If
+   * 'boolean', then return 0.0, otherwise parse the given string value to 'number'. If the input is
+   * null or not parseable, then an exception is thrown.
+   * 
+   * @param value the value to be parsed.
+   * @param type the type of the string value.
+   * @return the parsed value.
+   */
+  private static double parseScore(String value, String type) {
+    switch (type) {
+      case "number":
+        return Double.parseDouble(value);
+      case "boolean":
+        return 0.0;
+      default:
+        throw new IllegalArgumentException(String.format("Unknown type: %s", type));
+    }
+  }
+
+  /**
+   * Get the value if a configuration exists.
+   * 
+   * @param name The name of the configuration to find.
+   * @param configuration The list of configurations associated to OWASP Dependency Check plugin.
+   * @return The value of the configuration if found.
+   */
+  private static Optional<String> parameter(String name, Object configuration) {
+    if (configuration == null || !(configuration instanceof Xpp3Dom)) {
```

Comment:
I think the null-check if redundant here

```suggestion
    if (configuration instanceof Xpp3Dom == false) {
```

Reply:
Can this be written like this then?

`!(configuration instanceof Xpp3Dom)`

Reply:
Sure, it can. I just don't like `(!(`. But not a big deal. Feel free to write like this.